library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -633,21 +633,22 @@ public <T> T execute(final OperationChain<T> opChain) {
         for (final Operation operation : opChain.getOperations()) {
             operation.setOptions(opOptions);
 
-            if (operation instanceof Input) {
+            if (LOGGER.isDebugEnabled() && operation instanceof Input) {
                 Object input = ((Input) operation).getInput();
                 if (input != null) {
```

Comment:
Is this null check actually required?

Reply:
Potentially not now that it has moved from info to debug - so can be removed I think :)

Reply:
I was more thinking that as the only path that uses the `input` variable now checks if it is an instance of `MappedIterable` there is now no need for a null check (because you have removed the `toString()` call on the else branch).

Reply:
I think `LOGGER.isDebugEnabled()` should remain as this prevents needlessly iterating though the operation when nothing will ever be logged.